src/write_barrier.rs

```diff
@@ -165,6 +167,7 @@ impl<P, O> WriteBarrier<P, O> {
     /// the leader of a set of writers, returns a [`WriteGroup`] that contains
     /// them, `writer` included.
     pub fn enter<'a>(&self, writer: &'a mut Writer<P, O>) -> Option<WriteGroup<'_, 'a, P, O>> {
+        let _t = StopWatch::new(perf_context!(write_leader_wait_nanos));
```

Comment:
`ENGINE_WRITE_PREPROCESS_DURATION_HISTOGRAM` records the time from entering `write` to the data being written to pipe. And we treat it as the wait duration before.

But both two are incorrect. A better breakdown would be {compression} -> {wait} -> {writing write group} -> {apply}. Only the leader is aware of wait duration, so he must broadcast it to other writers (via an additional payload) so they can calculate their own write duration.

Reply:
I improve the breakdown now.

Now, before entering the write group, the write thread first snapshot its current perf context.

Only the write leader will record `write_leader_wait_nanos`, `log_write_nanos`, `log_rotate_nanos` and `log_sync_nanos`. And the diff at the leader will be broadcast to each `Writer`.

After the write group exists, each writer thread will recover its perf context snapshot and add the diff to it.